src/ess/livedata/core/job_manager.py

```diff
@@ -67,6 +74,53 @@ class JobFactory:
     def __init__(self, instrument: Instrument) -> None:
         self._instrument = instrument
 
+    def get_workflow_spec(self, workflow_id: WorkflowId) -> WorkflowSpec | None:
+        """Get the workflow specification for a given workflow ID."""
+        return self._instrument.workflow_factory.get(workflow_id)
+
+    def enrich_result(self, result: JobResult) -> JobResult:
+        """
+        Enrich a job result by setting human-readable names on output DataArrays.
+
+        This method looks up the workflow specification and uses the output metadata
+        to set the `.name` attribute on DataArrays based on their title field.
+
+        Parameters
+        ----------
+        result:
+            The job result to enrich with human-readable names.
+
+        Returns
+        -------
+        :
+            The enriched job result (modified in-place, but also returned).
+        """
+        # Skip enrichment if result has an error or no data
+        if result.error_message is not None or result.data is None:
+            return result
+
+        # Get workflow spec to access output metadata
+        workflow_spec = self.get_workflow_spec(result.workflow_id)
+        if workflow_spec is None or workflow_spec.outputs is None:
+            return result
+
+        def set_name_from_metadata(value, output_name: str) -> None:
+            """Set human-readable name on a value based on output metadata."""
+            if hasattr(value, 'name'):
```

Comment:
Would it make more sense to check for `DataArray` explicitly? What if `value` is some other type that happens to have a `name` attribute?